src/components/forms/DocumentUploadForm/shared/FileSelected/FileSelected.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+  <div>
+    <spinner v-if="isLoading" />
+    <component
+      :is="current"
+      v-else
+      :title="$t('components.uploadDocument.frontIsSelected')"
+      :image-content="imageContent"
+      :file-name="fileName"
+      @file-remove="$listeners['file-remove']"
+    />
+  </div>
+</template>
+
+<script>
+import UploadTitle from '@/components/forms/DocumentUploadForm/shared/UploadTitle';
+import Spinner from '@/components/common/Spinner';
+import FileAsTitle from './FileAsTitle';
+import FileAsImage from './FileAsImage';
+
+export default {
+  name: 'FileSelected',
+
+  props: {
+    file: {
+      type: File,
+      default: null,
+    },
+  },
+
+  data() {
+    return {
+      isLoading: true,
+      isFileAsTitle: true,
+      imageContent: null,
+      fileId: '',
+    };
+  },
+
+  computed: {
+    fileName() {
+      return this.file && this.file.name;
+    },
+
+    current() {
+      if (this.isLoading) {
+        return null;
+      }
+
+      if (this.isFileAsTitle) {
+        return FileAsTitle;
+      }
+
+      return FileAsImage;
+    },
+  },
+
+  watch: {
+    file: {
+      handler(file) {
+        if (!this.isImage(file)) {
+          this.isLoading = false;
+          this.isFileAsTitle = true;
+          return;
+        }
+
+        this.loadImageContent(file);
+      },
+      immediate: true,
+    },
+  },
+
+  methods: {
+    isImage(file) {
+      if (!file) return false;
+      return file.type.indexOf('image') === 0;
+    },
+
+    loadImageContent(file) {
+      this.isLoading = true;
+      this.isFileAsTitle = false;
+
+      this.fileId = `${file.name}-${file.size}`;
```

Comment:
This thing is not safe. For really safe cases we should compare file in base64 format. Do we really need to use that?
Error should always stop `isLoading`, but I don't understand, why we need render file title in some error cases?

Reply:
if file loading have error, it means that we have something wrong when loading file and we need to use title as fallback for this.
preview file - should not block upload image, it's just a common helper